File: 824-number-of-lines-to-write-string/number-of-lines-to-write-string.py
class Solution:
    def numberOfLines(self, widths: List[int], s: str) -> List[int]:
        count = 1
        temp = 0
        for i in range(len(s)):
            temp+=widths[ord(s[i])-97]
            if temp > 100:
                count+=1
                temp = widths[ord(s[i])-97]
        return [count,temp]

File: 824-number-of-lines-to-write-string/number-of-lines-to-write-string.js
/**
 * @param {number[]} widths
 * @param {string} s
 * @return {number[]}
 */
var numberOfLines = function(widths, s) {
    let count = 1
    let temp = 0
    for (let i=0;i<s.length;i++){
        let w = widths[s.charCodeAt(i)-97]
        if(w+temp > 100){
            count+=1
            temp = 0
        }
        temp+=w
    }
    return [count,temp]
};